frontend/lib/js/header/Header.js

```diff
@@ -1,18 +1,46 @@
-import React from 'react';
-import T     from 'i18n-react';
+import React           from 'react';
+import PropTypes       from 'prop-types';
+import T               from 'i18n-react';
+import { connect }     from 'react-redux';
+import { loadVersion } from './actions';
 
 class Header extends React.Component {
+  componentDidMount() {
+    this.props.loadVersion();
```

Comment:
Oh, and for parsing URL parameters, I had designated the Startup component to be "the only" component in the app to dispatch actions in componentDidMount -- of course there may be exceptions to every rule, but I guess it helps to keep an overview about what's going on when the app starts.

What do you think? Could you integrate this call into that component?

Reply:
I thought about it, but I am unsure if this the right way.
It makes sense to only load the concepts at startup and only once to minimize traffic.

Isn't it better, when the data is not too big, to load it where it is needed? So that every component is connected to it's data and handles loading it.
The Startup component, it makes it difficult to understand where the data is coming from.

Let's say I am only looking at Header, then I need to find the reducer to version, the action and then the component executing that action (Startup).
Also now I don't need Header anymore and I delete that component. It is not obvious that one needs to adapt Startup and remove the API call there as it is not needed anymore and even if, you need to refactor code more locations than, when calling inside the component itself.

But this is of course if you do not know that everything is loaded in Startup.
Would you simply fetch the version in `componentDidMount` of the Startup component?